### DISCUSSION TITLE
Feature Request: Secret to recognize credential_chain for GCS

### DISCUSSION THREAD

**Discussion by mattmartin14** [duckdb/duckdb#15501] [Ideas]
Duckdb had not too long ago released their new secrets manager, which is great and can easily pick up default environment credentials for AWS with something as simple as:

```python
cn.execute("""
    CREATE SECRET s1 (
           TYPE S3,
           PROVIDER CREDENTIAL_CHAIN
           )
""")
```

This approach is great because it removes the need to store hard credentials on your local workstation; instead you are reliant on a rotating session of temporary credentials that you had created with the AWS CLI.

However, when we try to do the same with GCS, it appears duckdb is unable to pick up the application default credentials, which you can place on your local workstation with the gcloud CLI as follows:

```bash
gcloud auth application-default login
```

As a current workaround, instead of using the duckdb secrets, we can simply use the fsspec python package and register the GCS file system. Example code to do this is as follows:

```python
import duckdb
from fsspec import filesystem
cn = duckdb.connect()
cn.register_filesystem(filesystem('gcs'))
```

After that is done, DuckDB can read data from gcs buckets. 

It would be great if the duckdb secrets were enhanced to capture the google default application credentials via the PROVIDER_CHAIN instead of the workaround above. These benefits would also translate to sessions that want to just use the duckdb cli instead of python.

I'm curious to hear other's thoughts on this matter?

**Comment by StanleyY** [comment]
Note that there are limitations with fsspec though. For example the lack of wildcard support mentioned here. https://github.com/duckdb/duckdb/issues/1403#issuecomment-2163174468

**Comment by StanleyY** [comment]
I wonder how difficult it would be to tweak this aws extension to support the gcp's ADC

https://github.com/duckdb/duckdb-aws

https://github.com/duckdb/duckdb/issues/4021
